static/src/stylesheets/module/nav/_brand-bar.scss

```diff
@@ -182,6 +182,13 @@
     }
 }
 
+.is-not-modern {
+    .brand-bar__item--more,
+    .brand-bar__item--edition {
+        display: none;
```

Comment:
Personal preference here, but I'm a fan of a general utility class e.g. `.u-needs-modern` or `.u-requires-javascript` which you apply to the markup directly for this kind of thing, rather than having to maintain this list in CSS.